mesonbuild/mesonmain.py

```diff
@@ -62,21 +62,31 @@ def setup_vsenv():
     if shutil.which('cl.exe'):
         return
 
-    bat_locator_bin = r'c:\Program Files (x86)\Microsoft Visual Studio\Installer\vswhere.exe'
-    if not os.path.exists(bat_locator_bin):
+    root = os.environ.get("ProgramFiles(x86)") or os.environ.get("ProgramFiles")
+    bat_locator_bin = pathlib.Path(root, 'Microsoft Visual Studio/Installer/vswhere.exe')
+    if not bat_locator_bin.exists():
         return
-    bat_json = subprocess.check_output([bat_locator_bin, '-latest', '-format', 'json'])
+    bat_json = subprocess.check_output(
+        [
+            str(bat_locator_bin),
+            '-latest',
+            '-prerelease',
+            '-requiresAny',
+            '-requires', 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64',
+            '-requires', 'Microsoft.VisualStudio.Workload.WDExpress',
+            '-products', '*',
+            '-format',
+            'json'
```

Comment:
While at it, according to some random stackoverflow I found, we don't need to parse the whole json, just add `-property installationPath` and it will print only the path.

Reply:
Ok.

Reply:
We are printing the version also
https://github.com/mesonbuild/meson/blob/dc5d3277ce599436b8763f3affb5f32671ed06d3/mesonbuild/mesonmain.py#L86
so I think it is fine to parse the json.